force-app/main/default/classes/AccountNamingSettingsMapper.cls

```diff
@@ -0,0 +1,112 @@
+* @group EDA Settings
+* @group-content ../../ApexDocContent/EdaSettings.htm
+* @description Class to act a mapper for Administrative Account Naming functionality.
+*/
+public virtual with sharing class AccountNamingSettingsMapper {
+    /**********************************************************************************
+    * @description Instance for Singleton Pattern
+    **********************************************************************************/
+    @TestVisible
+    private static AccountNamingSettingsMapper instance;
+
+    /**********************************************************************************
+    * @description Empty constructor is not public for Singleton Pattern
+    **********************************************************************************/
+    protected AccountNamingSettingsMapper() {}
+
+    /**********************************************************************************
+    * @description Static method to get the current instance for Singleton pattern\
+    * @return The instance of AccountNamingSettingsMapper.
+    **********************************************************************************/
+    public static AccountNamingSettingsMapper getInstance() {
+        if (instance == null) {
+            instance = new AccountNamingSettingsMapper();
+        }
+
+        return instance;
+    }
+
+    /**********************************************************************************
+    * @description Retrieves an Account Naming Settings model with naming the values for
+    * administrative and household account naming from hierarchy settings. 
+    * @return An AccountNamingSettingsModel
+    **********************************************************************************/
+    public virtual AccountNamingSettingsModel getAccountNamingSettingsModel() {
+        Hierarchy_Settings__c hierarchySettings = UTIL_CustomSettingsFacade.getSettings();
+
+        AccountNamingSettingsModel accountNamingSetttingsModel = 
+            new AccountNamingSettingsModel(
+                hierarchySettings.Admin_Account_Naming_Format__c, 
+                hierarchySettings.Household_Account_Naming_Format__c, 
+                hierarchySettings.Admin_Other_Name_Setting__c,
+                hierarchySettings.Household_Other_Name_Setting__c,
+                hierarchySettings.Automatic_Household_Naming__c
+            );
+
+       return accountNamingSetttingsModel;
+    }
+
+    /**********************************************************************************
+    * @description Retrieves an Account Naming Settings model with valid naming formats 
+    * for administrative accounts. 
+    * @return An AccountNamingFormatModel
+    **********************************************************************************/
+    public virtual AccountNamingFormatModel getAdministrativeAccountNamingFormatModel() {
+        Map<String, String> validAdminAccountNamingFormatsByName = new Map<String, String>();
```

Comment:
These are good. We might want to consider an enum for these at some point, but the string gives us simplicity for now.